### ISSUE TITLE
Testing fails during packaging (Debian Med)

### ISSUE THREAD

**Issue by shayandoust** [intake/intake#508]
Hello,

I am currently packaging intake[1] for Debian Med[2]. I am currently hitting an issue, specifically during the testing stage of the build (pytest). It seems like a lot of tests fail. Any help or suggestion would be much appreciated!

An example of the most common test fail is similarly as follows:

```
__________________________ test_second_load_timestamp __________________________

catalog_cache = <Intake catalog: catalog_caching>

    def test_second_load_timestamp(catalog_cache):
>       cat = catalog_cache['test_cache']

intake/catalog/tests/test_caching_integration.py:171: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
intake/catalog/base.py:387: in __getitem__
    return e()
intake/catalog/entry.py:77: in __call__
    s = self.get(**kwargs)
intake/catalog/local.py:280: in get
    plugin, open_args = self._create_open_args(user_parameters)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = name: test_cache
container: None
plugin: []
description: cache a csv file from the local filesystem
direct_access: forbid
user_parameters: []
metadata: 
args: 
  urlpath: {{ CATALOG_DIR }}/cache_data/states.csv
user_parameters = {}

    def _create_open_args(self, user_parameters):
        plugin = user_parameters.pop('plugin', None)
    
        md = self._metadata.copy() if self._metadata is not None else {}
        md['catalog_dir'] = self._catalog_dir
        if user_parameters.pop('cache', None) or self._cache:
            md['cache'] = user_parameters.pop('cache', None) or self._cache
        params = {'metadata': md,
                  'CATALOG_DIR': self._catalog_dir,
                  }
        params.update(self._open_args)
        if ('storage_options' not in params
                and self._filesystem is not None
                and self._filesystem.storage_options
                and _has_catalog_dir(params)):
            params['storage_options'] = self._filesystem.storage_options
        open_args = merge_pars(params, user_parameters, self._user_parameters,
                               getshell=self.getshell, getenv=self.getenv,
                               client=False)
    
        if len(self._plugin) == 0:
>           raise ValueError('No plugins loaded for this entry: %s\n'
                             'A listing of installable plugins can be found '
                             'at https://intake.readthedocs.io/en/latest/plugin'
                             '-directory.html .'
E                            ValueError: No plugins loaded for this entry: csv
E                            A listing of installable plugins can be found at https://intake.readthedocs.io/en/latest/plugin-directory.html .

intake/catalog/local.py:257: ValueError
```

I have attached a full log of the build to this issue. 

Kind regards,
Shayan Doust

[intake_sbuild_log.txt](https://github.com/intake/intake/files/4789180/intake_sbuild_log.txt)

[1]: https://salsa.debian.org/med-team/intake
[2]: https://www.debian.org/devel/debian-med/

**Comment by martindurant** [comment]
I would recommend using pip3 to install rather than `python3 setup.py build` - I suspect that the "entrypoints" in the setup script are now being respected, so the drivers appear not to be registered.

**Comment by shayandoust** [comment]
With packaging, the source code needs to be compiled (if applicable), installed within its correct location and tested. We use pybuild[1] to automate python packaging and file installations, so we wouldn't be able to use pip3. Pybuild does in fact execute "python3 setup.py build" during package building. Can you suggest any other possible solution? This is currently holding back the packaging of intake.

[1]: https://wiki.debian.org/Python/Pybuild

**Comment by martindurant** [comment]
It's hard to know from here what may be going wrong. Apparently the plugins registry did not get populated, which is something that should happen during package installation from the "entrypoints" metadata provided to pip. I suppose "python3 setup.py install" would do this too, but I am not certain; build alone would certainly not.

**Comment by martindurant** [comment]
(note that intake is pure-python and doesn't actually need a compilation step)

**Comment by shayandoust** [comment]
Thanks for your prompt replies. You're right. I must admit that I haven't really packaged any python-based software, so I simply opened a query with the Python packaging team within the mailing list and the issue was the missing egg-info file during pytest in the build directory, so I simply added "intake.egg-info" within pytest.testfiles so that pybuild could move this extra file. As a result, all tests are looking good!

Thanks again,
Shayan
